src/schema/v2/v2/index.ts

```diff
@@ -58,6 +58,17 @@ export const transformToV2 = (
     new FilterTypes(type => {
       return !opt.filterTypes.includes(type.name)
     }),
+    // Filter out `id` fields from Exchange types
+    new FilterFields(
+      (type, field) =>
+        !type.name.startsWith("Commerce") ||
+        // TODO: These two exchange types are only meant for stitching and
+        //       should simply be filtered out after stitching, but I don't want
+        //       to make such a large sweeping change right now.
+        type.name === "CommercePartner" ||
+        type.name === "CommerceUser" ||
+        field.name !== "id"
+    ),
```

Comment:
In the long run, this shouldn’t remain here, in fact this entire file should be retired, but as stated I will do that after the v2 work is ready for Emission as it requires a bit more thoughtful work around stitching and v1.